### ISSUE TITLE
[BUG] unasync'd HiveSession.updateData generates warnings and exceptions on lock acquisition / release

### ISSUE THREAD

**Issue by gormanb** [Pyhass/Pyhiveapi#74]
**Describe the bug**
When `pyhiveapi` is [packaged via `unasync`](https://github.com/Pyhass/Pyhiveapi/blob/05d5ae8a664160e2056906b24bba44e80729b52a/setup.py#L23-L33), it removes the `await` from [this `asyncio.Lock` acquisition in `HiveSession.updateData`](https://github.com/Pyhass/Pyhiveapi/blob/05d5ae8a664160e2056906b24bba44e80729b52a/pyhiveapi/apyhiveapi/session.py#L328) but otherwise leaves the `asyncio.Lock` in place. As a result, calling `updateData` generates the following warning:
```
/path/to/session.py:328: RuntimeWarning: coroutine 'Lock.acquire' was never awaited
  self.updateLock.acquire()
```
... and can throw the following exception if it attempts to [release the lock](https://github.com/Pyhass/Pyhiveapi/blob/05d5ae8a664160e2056906b24bba44e80729b52a/pyhiveapi/apyhiveapi/session.py#L335) without having acquired it:
```
[1/13/2024, 2:20:09 PM] Error: Python exception: Lock is not acquired.
```

**Expected behavior**
Calling `HiveSession.updateData` succeeds without spurious warning or errors.

**Additional context**
[I have a patch here](https://github.com/Pyhass/Pyhiveapi/compare/master...gormanb:Pyhiveapi:v0.5.16.1) which simply adds an `unasync` rule to convert the `asyncio.Lock` in `session.py` to `threading.Lock` instead. No other code is affected by this change. I can open a PR, if you think this is an appropriate solution?

**Comment by Khole-FG** [comment]
@gormanb  Yes please raise a pr for this

**Comment by gormanb** [comment]
> @gormanb Yes please raise a pr for this

Done!

**Comment by gormanb** [comment]
Fixed by PR #75 in commit b3ee963.
